### PR TITLE
refactor: removed duplicate definitions of storage prefixes

### PR DESCRIPTION
**Pull request by nimrod-starkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/committer/54)

### CODE REVIEW

File: crates/committer/src/storage/storage_trait.rs [hunk 1]
```diff
@@ -29,32 +29,38 @@ pub(crate) trait Storage {
...
+            Self::InnerNode => b"patricia_node",
+            Self::StorageLeaf => b"starknet_storage_leaf",
+            Self::StateTreeLeaf => b"contract_state",
+            Self::CompiledClassLeaf => b"contract_class_leaf",
         }
     }
 }
 
 impl StorageKey {
     pub(crate) fn with_prefix(&self, prefix: StoragePrefix) -> Self {
```
Comment (AvivYossef-starkware) [inline crates/committer/src/storage/storage_trait.rs:51]: In my opinion, this logic is confusing and will run slower than using create_db_key

File: crates/committer/src/storage/storage_trait.rs [hunk 2]
```diff
@@ -29,32 +29,38 @@ pub(crate) trait Storage {
...
-        let mut prefix = prefix.to_bytes().to_vec();
+        let mut prefix = [prefix.to_bytes().to_vec(), b":".to_vec()].concat();
         prefix.extend(&self.0);
         StorageKey(prefix)
     }
+
+    pub(crate) fn from_prefix_and_suffix(prefix: StoragePrefix, suffix: &[u8]) -> Self {
```
Comment (AvivYossef-starkware) [inline crates/committer/src/storage/storage_trait.rs:57]: Same as comment in line 44

File: crates/committer/src/storage/storage_trait.rs [hunk 3]
```diff
@@ -29,32 +29,38 @@ pub(crate) trait Storage {
 }
 
 pub(crate) enum StoragePrefix {
-    PatriciaNode,
```
Comment (AvivYossef-starkware) [inline crates/committer/src/storage/storage_trait.rs:32]: Was it a mistake?

**Review by nimrod-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/54)* status: 0 of 5 files reviewed, 3 unresolved discussions (waiting on @AvivYossef-starkware and @dorimedini-starkware)

___
*[`crates/committer/src/storage/storage_trait.rs` line 0 at r1](https://reviewable.io/reviews/starkware-libs/committer/54#gh-1573675097:-NvzqxgMDPMRhyyAz4QR:b-ysumu6) ([raw file](https://github.com/starkware-libs/committer/blob/f263a47426c1fd71668252b0954d13f1546be4bc/crates/committer/src/storage/storage_trait.rs#L0)):*
<details><summary><i>Previously, AvivYossef-starkware wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map [this GitHub inline comment thread](https://github.com/starkware-libs/committer/pull/54#discussion_r1573675097) to the right spot — sorry!)*

Was it a mistake?
</blockquote></details>

no, i renamed it (it was `PatriciaNode` and now it's `InnerNode`)

**Review by AvivYossef-starkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/54)* status: 0 of 5 files reviewed, 4 unresolved discussions (waiting on @amosStarkware, @dorimedini-starkware, and @nimrod-starkware)

___
*[`crates/committer/src/storage/storage_trait.rs` line 40 at r2](https://reviewable.io/reviews/starkware-libs/committer/54#-Nw--2Zq7nmz5LjE7Dx5:-Nw--2Zq7nmz5LjE7Dx6:b-armfv3) ([raw file](https://github.com/starkware-libs/committer/blob/87c5b5360587b701509b5313895c0c6577241009/crates/committer/src/storage/storage_trait.rs#L40)):*
> ```XML
> /// Describes a storage prefix as used in Aerospike DB.
> impl StoragePrefix {
>     pub(crate) fn to_bytes(&self) -> &[u8] {
> ```

In the current implementation, the byte slices are created every time the `to_bytes` method is called. we lose running time. I think you need to use a static field for the enum
___
*[`crates/committer/src/storage/storage_trait.rs` line 52 at r2](https://reviewable.io/reviews/starkware-libs/committer/54#-Nw-285Q62fc5yZaIGHf:-Nw-285Q62fc5yZaIGHg:bqdyfey) ([raw file](https://github.com/starkware-libs/committer/blob/87c5b5360587b701509b5313895c0c6577241009/crates/committer/src/storage/storage_trait.rs#L52)):*
> ```XML
> #[allow(dead_code)]
> impl StorageKey {
>     pub(crate) fn with_prefix(&self, prefix: StoragePrefix) -> Self {
> ```

do you use it?

**Review by nimrod-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/54)* status: 0 of 5 files reviewed, 4 unresolved discussions (waiting on @amosStarkware, @AvivYossef-starkware, and @dorimedini-starkware)

___
*[`crates/committer/src/storage/storage_trait.rs` line 51 at r1](https://reviewable.io/reviews/starkware-libs/committer/54#gh-1573674953:-Nw-5UTUEidkFJIzNciw:b-896fix) ([raw file](https://github.com/starkware-libs/committer/blob/f263a47426c1fd71668252b0954d13f1546be4bc/crates/committer/src/storage/storage_trait.rs#L51)):*
<details><summary><i>Previously, AvivYossef-starkware wrote…</i></summary><blockquote>

In my opinion, this logic is confusing and will run slower than using create_db_key
</blockquote></details>

Done.
___
*[`crates/committer/src/storage/storage_trait.rs` line 57 at r1](https://reviewable.io/reviews/starkware-libs/committer/54#gh-1573675081:-Nw-5Ujy7n71vXRvKNW4:b-896fix) ([raw file](https://github.com/starkware-libs/committer/blob/f263a47426c1fd71668252b0954d13f1546be4bc/crates/committer/src/storage/storage_trait.rs#L57)):*
<details><summary><i>Previously, AvivYossef-starkware wrote…</i></summary><blockquote>

Same as comment in line 44
</blockquote></details>

Done.
___
*[`crates/committer/src/storage/storage_trait.rs` line 40 at r2](https://reviewable.io/reviews/starkware-libs/committer/54#-Nw--2Zq7nmz5LjE7Dx5:-Nw-72RW9Sjwg8G5rr_s:b-jtaas5) ([raw file](https://github.com/starkware-libs/committer/blob/87c5b5360587b701509b5313895c0c6577241009/crates/committer/src/storage/storage_trait.rs#L40)):*
<details><summary><i>Previously, AvivYossef-starkware wrote…</i></summary><blockquote>

In the current implementation, the byte slices are created every time the `to_bytes` method is called. we lose running time. I think you need to use a static field for the enum
</blockquote></details>

good point. changed the return value to `&'static [u8]`
___
*[`crates/committer/src/storage/storage_trait.rs` line 52 at r2](https://reviewable.io/reviews/starkware-libs/committer/54#-Nw-285Q62fc5yZaIGHf:-Nw-5I0D3_Xl3ATreFE4:b-5te6zm) ([raw file](https://github.com/starkware-libs/committer/blob/87c5b5360587b701509b5313895c0c6577241009/crates/committer/src/storage/storage_trait.rs#L52)):*
<details><summary><i>Previously, AvivYossef-starkware wrote…</i></summary><blockquote>

do you use it?
</blockquote></details>

yes, currently at tests only.
changed it to `#[cfg(test)]`

**Review by dorimedini-starkware — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3, 7 of 7 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/54#-:-NwE1zIR5KBK8-EY2iP4:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @amosStarkware and @nimrod-starkware)

___
*[`crates/committer/src/patricia_merkle_tree/filled_tree/node_serde.rs` line 86 at r4](https://reviewable.io/reviews/starkware-libs/committer/54#-NwE1PStEUAsZIRJlorW:-NwE1PStEUAsZIRJlorX:b1nxvpe) ([raw file](https://github.com/starkware-libs/committer/blob/c147c8bf316c0404e47c5eab61e196bfb46cc46a/crates/committer/src/patricia_merkle_tree/filled_tree/node_serde.rs#L86)):*
> ```XML
>     }
> 
>     /// Returns the db key of the filled node - [prefix + b":" + suffix].
> ```

remove the implementation detail from the docstring pls



_Suggestion:_
```XML
/// Returns the db key of the filled node.
```
___
*[`crates/committer/src/storage/storage_trait.rs` line 55 at r4](https://reviewable.io/reviews/starkware-libs/committer/54#-NwE1jkPBfE2DCwyAHEL:-NwE1jkPBfE2DCwyAHEM:b-1gey5a) ([raw file](https://github.com/starkware-libs/committer/blob/c147c8bf316c0404e47c5eab61e196bfb46cc46a/crates/committer/src/storage/storage_trait.rs#L55)):*
> ```XML
>         create_db_key(prefix, &self.0)
>     }
> }
> ```

is this needed? it is strange IMO, putting test-only code in a non-test module... consider moving this to a test module (and dropping the `cfg(test)`)



_Code quote:_
```XML
#[cfg(test)]
impl StorageKey {
    pub(crate) fn with_prefix(&self, prefix: StoragePrefix) -> Self {
        create_db_key(prefix, &self.0)
    }
}
```

**Review by nimrod-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/54)* status: all files reviewed, 1 unresolved discussion (waiting on @amosStarkware and @dorimedini-starkware)

___
*[`crates/committer/src/storage/storage_trait.rs` line 55 at r4](https://reviewable.io/reviews/starkware-libs/committer/54#-NwE1jkPBfE2DCwyAHEL:-NwE5d4k4WHzoxc_G7MO:b-oicyv6) ([raw file](https://github.com/starkware-libs/committer/blob/c147c8bf316c0404e47c5eab61e196bfb46cc46a/crates/committer/src/storage/storage_trait.rs#L55)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

is this needed? it is strange IMO, putting test-only code in a non-test module... consider moving this to a test module (and dropping the `cfg(test)`)
</blockquote></details>

removed it.

**Review by dorimedini-starkware — APPROVED** [review]
> Reviewed 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/54)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @amosStarkware)

**Review by AvivYossef-starkware — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/54#-:-NwE8h0BE_efs26K6dcX:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @amosStarkware)

**Pull request merged**